src/gpgme.cc

```diff
@@ -183,15 +179,15 @@ decrypted_stream_t::decrypted_stream_t(path& filename)
 
   file = open_file(filename);
   auto enc_d = setup_cipher_buffer(file);
-  dec_d = decrypt(enc_d);
-  rewind(dec_d.get());
+  auto dec_d = decrypt(enc_d).get();
```

Comment:
I guess my previous comment got lost, because updates to comments do not get communicated via e-mail.
This still doesn't work. `dec_d` is a class member and you just made it a local variable. We must keep the shared pointer to the decrypted Data for the life time of the stream. So, please remove the `auto dec_d` declaration again and just keep the assignment.

Reply:
It should look like before, i.e.:
```cpp
dec_d = decrypt(enc_d);
dec_d.get()->rewind();
...
...(new data_streambuffer_t(*dec_d.get()));
```

Reply:
Thank you for you thorough review and helpful comments, I should've avoided to do needles code-optimisations… 🤷

Anyhow I think the PR is now as you requested it to be.